test/server/echo.go

```diff
@@ -120,3 +143,61 @@ func main() {
 	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
 	<-sigs
 }
+
+func setResponseFromCodes(request *http.Request, response http.ResponseWriter) error {
+	responseCodes := request.FormValue("codes");
+
+	codes, err := validateCodes(responseCodes)
+	if err != nil {
+		return err
+	}
+
+	// Keep a cursor for each sequence of response codes
+	position, ok := state[responseCodes]
```

Comment:
I didn't want to add a dependency because the output of this server when invoked by multiple callers or when implemented with multiple instances is not deterministic.  Therefore it doesn't matter if state gets corrupt.

If you disagree let me know and I will add the mutex.